nodejs/week3/warmup/calculator.js

```diff
@@ -0,0 +1,75 @@
+const e = require("express");
+const express = require("express");
+const router = express.Router();
+// const knex = require("../database");
+// const { request, response } = require("express");
+
+router.get("/", async(request, response) => {
+    try {
+        console.log("this is a calculator");
+        response.send("Let's get some calculations done");
+    } catch (error) {
+        throw error;
+    }
+});
+
+// GET section
+router.get("/add", (request, response) => {
+    const result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) + Number(b), 0);
+    response.send(`This is the result ${result}`);
+});
+
+router.get("/sub", (request, response) => {
+    const result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) - Number(b));
+    response.send(`This is the result ${result}`);
+});
+
+router.get("/multiply", (request, response) => {
+    let result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) * Number(b));
+    response.send(`This is the result ${result}`);
+});
+
+router.get("/divide", (request, response) => {
+    let result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) / Number(b));
+    response.send(`This is the result ${result}`);
+});
+
+// POST section
+
+router.post("/add", (request, response) => {
+    const result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) + Number(b), 0);
```

Comment:
You are using reduce which is a nice way to do this :+1: 
Also, one more thing you could do, optionally, to make your code more re-usable is to create a separate function for adding/subtracting etc.. and call them both in GET and POST requests.
That way, you avoid copy pasting the same code :)